crates/pop-contracts/src/utils/metadata.rs

```diff
@@ -167,6 +269,38 @@ pub async fn fetch_contract_storage(
 	account: &str,
 	rpc_url: &Url,
 	path: &Path,
+) -> anyhow::Result<String> {
+	fetch_contract_storage_with_param(storage, account, rpc_url, path, None).await
+}
+
+/// Fetches and decodes a storage value from a deployed smart contract,
+/// with optional filtering for mappings.
+///
+/// This function retrieves the value of a storage item from a deployed smart contract.
+/// For regular storage items, it returns the decoded value.
+/// For mapping types (Mapping<K,V>), it can either return all key-value pairs or filter for a
+/// specific key if provided.
+///
+/// # Arguments
+/// * `storage` - Storage item descriptor containing key and type information
+/// * `account` - Contract address as string (typically in H160 format)
+/// * `rpc_url` - URL of the RPC endpoint to connect to
+/// * `path` - Path to contract artifacts for metadata access
+/// * `mapping_key` - Optional key string for filtering mapping entries. Only used if the storage
+///   item is a Mapping<K,V>. The key string must be compatible with the mapping's key type K.
+///
+/// # Returns
```

Comment:
Just nitpick, for consistency we don't usually have the return in the comments, only the arguments